participant/prisma/seed.ts

```diff
@@ -1,33 +1,86 @@
 import { PrismaClient } from "@prisma/client";
+import {
+  findLocalAgency,
+  findSubmission,
+  upsertLocalAgency,
+  upsertSubmission,
+  upsertSubmissionForm,
+} from "app/utils/db.server";
+import seedAgencies from "public/data/local-agencies.json";
+import seedSubmissions from "public/data/submissions.json";
+
+// Define a bunch of types to make typescript happy for the case where
+// seedSubmissions is empty. Otherwise, typescript is usually able to
+// infer all the types.
+export type NameFormType = {
+  firstName: string;
+  lastName: string;
+  preferredName?: string;
+};
+
+export type SeedDataType = {
+  [key: string]: string;
+};
+
+export type SeedSubmissionFormsType = {
+  name: NameFormType;
+  [key: string]: SeedDataType | SeedDataType[];
+};
+
+export type SeedAgencySubmissionsType = {
+  submissionId: string;
+  forms: SeedSubmissionFormsType;
+};
+
+export type SeedSubmissionsType = {
+  [key: string]: SeedAgencySubmissionsType[];
+};
 
 const prisma = new PrismaClient();
-const agencyData = [
-  {
-    urlId: "gallatin",
-    name: "Gallatin WIC",
-  },
-  {
-    urlId: "missoula",
-    name: "Missoula WIC",
-  },
-];
 // Any interactions with Prisma will be async
 // eslint-disable-next-line @typescript-eslint/require-await
 async function seed() {
   // Put the actions you need to take to seed the databse here.
   // You can access relations as normal here
   // const users = await prisma.user.findMany(); // (for example)
-  const existingAgencyRecords = await prisma.localAgency.count();
-  if (existingAgencyRecords != agencyData.length) {
-    await prisma.localAgency.createMany({
-      data: agencyData,
-      skipDuplicates: true,
-    });
-    console.log(`Database has been seeded. 🌱`);
-  } else {
-    console.log(
-      `Skipping localAgency seed; found ${existingAgencyRecords} agencies 🪴`
-    );
+
+  // Seed local agencies.
+  for (const seedAgency of seedAgencies) {
+    const localAgency = await findLocalAgency(seedAgency.urlId);
+    if (!localAgency || localAgency.name != seedAgency.name) {
+      await upsertLocalAgency(seedAgency.urlId, seedAgency.name);
+      console.log(`Seeding localAgency: ${seedAgency.urlId} 🌱`);
+    }
+  }
```

Comment:
In remix eligibility, I do the length check against the seed lengths to avoid having to check every record..
if they're not very long, probably not worth optimizing though

(that is, if the length of seedAgencies is the same as the prisma.localAgency.count(), don't seed)

Reply:
Yah I saw that. The problem is that I'm actually changing the existing seed data records in this PR, so I wanted those changes to propagate and for us to have the ability to continue to make changes to the seed data if needed. But yah, I agree this would be a problem if this was a lot of records.